File: functions/libsodium/variants/go/2_b64_char_to_byte_2.go
package main

func b64CharToByte(c int32) uint32 {
	switch {
	case c >= 'A' && c <= 'Z':
		return uint32(c - 'A')
	case c >= 'a' && c <= 'z':
		return uint32(c - 'a' + 26)
	case c >= '0' && c <= '9':
		return uint32(c - '0' + 52)
	case c == '+':
		return 62
	case c == '/':
		return 63
	default:
		return 0
	}
}


func main(){
    b64CharToByte(0)
}

File: functions/libsodium/variants/c/2_b64_char_to_byte_2.c
#include <stdint.h>

static unsigned int b64_char_to_byte(int c)
{
    switch (c) {
        case '+': return 62;
        case '/': return 63;
        default:
            if (c >= 'A' && c <= 'Z') return c - 'A';
            if (c >= 'a' && c <= 'z') return c - 'a' + 26;
            if (c >= '0' && c <= '9') return c - '0' + 52;
            return 0;
    }
}


int main(){
    b64_char_to_byte(0);
}